src/efftester.ml

```diff
@@ -1210,15 +1381,16 @@ module Shrinker = struct
     | _ -> Iter.empty
   ;;
 
-  let ( <+> ) = Iter.( <+> )
-
   let rec term_shrinker term =
+    let ( <+> ) = Iter.( <+> ) in
     match term with
     | Lit l -> shrink_lit l
     | Variable (t, _) ->
       (match create_lit t with
       | Some c -> Iter.return c
       | _ -> Iter.empty)
+    | Constructor _ as c -> Iter.return c
+    | PatternMatch _ -> failwith "FIXME: term_shrinker: pat match not implemented"
```

Comment:
If a pattern-matching has a branch that binds no variables (`| None -> e`, or `| Some(None) -> e`, etc.), you can always shrink the whole matching by reducing to that branch only. Otherwise, you should just shrink all parts recursively. (We could do more sophisticated things by shrinking complex patterns into simpler patterns, etc., but that would be more work to implement and I think not terribly useful.)